Auth authlib ALTERNATIVE API

kajarenc opened the pull request:
## Describe your changes

## GitHub Issue Link (if applicable)

## Testing Plan

- Explanation of why no additional tests are needed
- Unit Tests (JS and/or Python)
- E2E Tests
- Any manual testing needed?

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

kajarenc committed:
move login / logout outside of st.experimental_user

kajarenc committed:
migrate playwright tests

sfc-gh-kmcgrady reviewed (approved):
Assuming, this is the preferred API, the changes look good.

kajarenc committed:
Merge branch 'auth-authlib-experiments' into auth-authlib-alternative-api

# Conflicts:
#	lib/streamlit/user_info.py

kajarenc committed:
Merge branch 'auth-authlib-experiments' into auth-authlib-alternative-api

# Conflicts:
#	e2e_playwright/auth.py
#	lib/streamlit/user_info.py
#	lib/tests/streamlit/user_info_test.py

kajarenc committed:
Merge branch 'auth-authlib-experiments' into auth-authlib-alternative-api

kajarenc committed:
define provider as an optional argument for st.login()

kajarenc committed:
remove asterisks from exception text

kajarenc committed:
Auth authlib lazy authlib import (#10113)

## Describe your changes

## GitHub Issue Link (if applicable)

## Testing Plan

- Explanation of why no additional tests are needed
- Unit Tests (JS and/or Python)
- E2E Tests
- Any manual testing needed?

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this
project are made under the Apache 2.0 license.

jrieke committed:
Change wording for auth exception messages (#10110)

## Describe your changes

## GitHub Issue Link (if applicable)

## Testing Plan

- Explanation of why no additional tests are needed
- Unit Tests (JS and/or Python)
- E2E Tests
- Any manual testing needed?

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this
project are made under the Apache 2.0 license.

---------

Co-authored-by: Johannes Rieke <johannes.rieke@snowflake.com>
Co-authored-by: Karen Javadyan <kajarenc@gmail.com>

kajarenc committed:
rename additional field to "is_logged_in"

kajarenc committed:
Make cookie_secret in secrets.toml mandatory for Auth

kajarenc committed:
add `is_logged_in` field default to False if auth section exists in secrets

kajarenc committed:
fix failing tests, and new test for missing cookie_secret

kajarenc committed:
check Authlib version semantically correct way
